com.ibm.wala.core/src/com/ibm/wala/classLoader/ClassLoaderImpl.java

```diff
@@ -242,58 +242,45 @@ public void clear() {
    * Set up the set of classes loaded by this object.
    */
   @SuppressWarnings("unused")
-  private void loadAllClasses(Collection<ModuleEntry> moduleEntries, Map<String, Object> fileContents) {
+  private void loadAllClasses(Collection<ModuleEntry> moduleEntries) {
     for (Iterator<ModuleEntry> it = moduleEntries.iterator(); it.hasNext();) {
       ModuleEntry entry = it.next();
       if (!entry.isClassFile()) {
         continue;
       }
-
       String className = entry.getClassName().replace('.', '/');
-
       if (DEBUG_LEVEL > 0) {
         System.err.println("Consider " + className);
       }
-
-      if (exclusions != null && exclusions.contains(className)) {
+      try {
+        ShrikeClassReaderHandle entryReader = new ShrikeClassReaderHandle(entry);
+        String classNameInternal = entryReader.get().getName();
         if (DEBUG_LEVEL > 0) {
-          System.err.println("Excluding " + className);
+          System.err.println("Load class " + classNameInternal);
         }
-        continue;
-      }
-
-      ShrikeClassReaderHandle entryReader = new ShrikeClassReaderHandle(entry);
-
-      className = "L" + className;
-      if (DEBUG_LEVEL > 0) {
-        System.err.println("Load class " + className);
-      }
-      try {
-        TypeName T = TypeName.string2TypeName(className);
-        if (loadedClasses.get(T) != null) {
+        if (exclusions != null && exclusions.contains(classNameInternal)) {
+          if (DEBUG_LEVEL > 0) {
+            System.err.println("Excluding " + classNameInternal);
+          }
+          continue;
+        }
+        // if the internal name does not correspond to the path we add a warning but still add the class
+        if (!classNameInternal.equals(className)) {
+          Warnings.add(InvalidClassFile.create(className));
+        }
+        classNameInternal = "L" + classNameInternal;
+        TypeName internalT = TypeName.string2TypeName(classNameInternal);
+        // add the class if it is not present yet
+        if (loadedClasses.get(internalT) != null) {
           Warnings.add(MultipleImplementationsWarning.create(className));
-        } else if (parent != null && parent.lookupClass(T) != null) {
+        } else if (parent != null && parent.lookupClass(internalT) != null) {
           Warnings.add(MultipleImplementationsWarning.create(className));
         } else {
-          // try to read from memory
-          ShrikeClassReaderHandle reader = entryReader;
-          if (fileContents != null) {
```

Comment:
So, the argument fileContents does not exist anymore. Can you clarify what this removed logic was doing and why we don't need it anymore?

Reply:
The argument fileContents was being called always with 'null'. Have a look at the lines 515-526 of this same file.
So that logic was basically dead code.

Reply:
Ok, I see...